File: 88. Merge Sorted Array: O((m+n)log(m+n)).rb
# 88. Merge Sorted Array
# https://leetcode.com/problems/merge-sorted-array/description/?envType=study-plan-v2&envId=top-interview-150

# @param {Integer[]} nums1
# @param {Integer} m
# @param {Integer[]} nums2
# @param {Integer} n

# Trivial O((m + n) * log(m + n)) algorithm for merging two sorted arrays into sorted array:
# @return {Void} Do not return anything, modify nums1 in-place instead.
def merge(nums1, m, nums2, n)
    m = nums1.length
    n = nums2.length

    # Replace the last zeroes of `nums1` with `nums2`:
    for i in 0...n do
        nums1[m - i - 1] = nums2[i]
    end
    
    # Sort number array in O(xlogx):
    nums1.sort!
end

File: 88. Merge Sorted Array: O((m+n)log(m+n)).py
# 88. Merge Sorted Array
# https://leetcode.com/problems/merge-sorted-array/description/?envType=study-plan-v2&envId=top-interview-150

# Trivial O((m + n) * log(m + n)) algorithm for merging two sorted arrays into sorted array:
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """

        m = len(nums1)
        n = len(nums2)

        # Replace the last zeroes of `nums1` with `nums2`:
        for i in range(n):
            nums1[m - i - 1] = nums2[i]
        
        # Sort number array in O(xlogx):
        nums1.sort()
